modules/contact/include/userobjects/WeightedGapUserObject.h

```diff
@@ -105,6 +102,9 @@ class WeightedGapUserObject : public MortarUserObject
   /// A map from node to weighted gap and normalization (if requested)
   std::unordered_map<const DofObject *, std::pair<ADReal, Real>> _dof_to_weighted_gap;
 
+  /// A map from node to unscaled weighted gap
+  std::unordered_map<const DofObject *, ADReal> _dof_to_real_weighted_gap;
```

Comment:
So I think that we've set up error checking to ensure that we're always nodal. I think we should replace our `DofObject` map keys by `Node` map keys. This should reduce the number of `static_cast` we do

Reply:
Doing this would consequences in other parts of the code. For example, in the parallel communication or other mortar contact objects that have not been migrated to the user object design. I'd suggest we add this as a task to a maintenance issue and try to get to that by the end of the FY.

Reply:
Ok please add it to a maintenance issue. @dschwen told me this morning you guys would largely like this PR merged as-is

Reply:
https://github.com/idaholab/moose/issues/23889#issue-1646302606

I am going to push changes that address most of your comments, but not the couple of them that involve deeper design changes.